### PR TITLE
SALTO-2478 Change salesforce adapter config default timeout

### PR DESCRIPTION
**Pull request by Uri-Bar**
_Change salesforce adapter config default timeout_

---
_Additional context for reviewer_:

* after consultation with Tamir, changing configuration shouldn't break any tests, so I only changed the tests that rely on the fact that we retry 5 times.
---
_Release Notes_: 
Salesforce-adapter:

* by lowering the number of time we attempt to do a retry from 5 to 3 and the duration of each try ,we might make the adapter more stable
---

_User Notifications_: 
None

### CODE REVIEW

File: packages/salesforce-adapter/test/client.test.ts [hunk 1]
```diff
@@ -312,7 +312,6 @@ describe('salesforce client', () => {
     let testClient: SalesforceClient
     let testConnection: MockInterface<Connection>
     let nullFailingImplementation: Metadata['list']
-    let rangeErrorFailingImplementation: Metadata['list']
```
Comment (noamhammer) [inline packages/salesforce-adapter/test/client.test.ts:315]: why did you remove this part of the code?

**Review by noamhammer — APPROVED** [review]

File: packages/salesforce-adapter/test/client.test.ts (hunk 1 above)
Comment (Uri-Bar) [reply packages/salesforce-adapter/test/client.test.ts:315]: not being used after the change
